### PR TITLE
updated Wrapper and Swap contracts, added Index report as well

### PR DESCRIPTION
**Pull request by poiresel**
## Description

Update swap, wrapper, and type addresses for reports.
Added first pass of index report

### CODE REVIEW

File: protocols/index/reports/SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+** Ownable.sol
+```
+
+_\*\* OpenZeppelin contract_
+
+#### Public and external functions (non-getter functions)
+
+| Function    | Source      | Visibility | Params                                                             | Payable |
+| :-------    | :---------- | :--------- | :----------------------------------------------------------------- | :------ |
+| setLocator  | Index.sol   | external   | `address _user, uint256 _score, bytes32 _locator`                  | no      |
```
Comment (dmosites) [inline protocols/index/reports/SECURITY.md]: `bytes32 _data`

File: protocols/index/reports/SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+## Contracts
+
+```
+Index.sol
+** Ownable.sol
+```
+
+_\*\* OpenZeppelin contract_
+
+#### Public and external functions (non-getter functions)
```
Comment (dmosites) [inline protocols/index/reports/SECURITY.md]: We should include `fetchLocators` here

File: protocols/index/reports/SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+#### 4. Each address can have either 0 or 1 locators on the index at any time - and therefore calling setLocator() twice does not disrupt the ordering of the locator linked list.
+
+- When the contract is created, the list is initialised as empty, and therefore no Locators exist for any user addresses.
+- When a new Locator is being set, if the address already has a Locator, the transaction reverts with LOCATOR_ALREADY_SET ensuring addresses cannot have more than 1 Locator on the index.
+- **This invariant currently holds as-is.**
+
+#### 5. Calling unsetLocator() twice does not disrupt ordering of the locator linked list.
+
+- At the beginning of unsetLocator, the function checks whether the specified address actually has an address. If they do not, the function returns false and stopped executing
```
Comment (ejwessel) [inline protocols/index/reports/SECURITY.md]: change to "At the beginning of unsetLocator, the function checks whether the specified address actually has an address. If they do not, the function returns false and stops executing"

File: protocols/index/reports/SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+- unsetLocator therefore only executes for addresses that have a Locator at that moment in time
+- Calling unsetLocator twice would (1) remove the Locator and then (2) return false and stop executing
```
Comment (ejwessel) [inline protocols/index/reports/SECURITY.md]: Calling unsetLocator twice would (1) remove the Locator and then (2) return false and stop executing

File: protocols/index/reports/SECURITY.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+- The locatorsLinkedList is modified in 3 places in the contract. In all places, there is a modifier onlyOwner at the beginning:
+    - First, the constructor. The caller of the constructor of the contract becomes the contract’s owner by definition of OpenZeppelin’s Ownable contract.
+    - Second in unsetLocator. This function is marked as onlyOwner, meaning that it can only be called successfully by the contract's owner.
+    - Third in setLocator. This function is marked as onlyOwner, meaning that it can only be called successfully by the contract's owner.
+- **This invariant currently holds as-is.**
+
+
+#### 3. Linked list ordering by score must be preserved at all times (Highest -> Lowest).
+
+a. Inserting Locators preserves the ordering of the list.
```
Comment (ejwessel) [inline protocols/index/reports/SECURITY.md]: this will be modified/updated with the fixing of the index.

Comment (poiresel) [reply protocols/index/reports/SECURITY.md]: we're not fixing it, it's good.

File: protocols/index/reports/SECURITY.md (hunk 3 above)
Comment (poiresel) [reply protocols/index/reports/SECURITY.md]: done!

File: protocols/index/reports/SECURITY.md (hunk 4 above)
Comment (poiresel) [reply protocols/index/reports/SECURITY.md]: removed!

File: protocols/index/reports/SECURITY.md [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+#### 4. Each address can have either 0 or 1 locators on the index at any time - and therefore calling setLocator() twice does not disrupt the ordering of the locator linked list.
+
+- When the contract is created, the list is initialised as empty, and therefore no Locators exist for any user addresses.
+- When a new Locator is being set, if the address already has a Locator, the transaction reverts with LOCATOR_ALREADY_SET ensuring addresses cannot have more than 1 Locator on the index.
+- **This invariant currently holds as-is.**
+
+#### 5. Calling unsetLocator() twice does not disrupt ordering of the locator linked list.
+
+- At the beginning of unsetLocator, the function checks whether the specified address actually has an address. If they do not, the function returns false and stops executing
+"At the beginning of unsetLocator, the function checks whether the specified address actually has an address. If they do not, the function returns false and stops executing
```
Comment (ejwessel) [inline protocols/index/reports/SECURITY.md]: duplicated with the line above.

**Pull request merged**